include/irr/asset/IDescriptorSetLayout.h

```diff
@@ -33,6 +33,27 @@ class IDescriptorSetLayout
         uint32_t count;
         E_SHADER_STAGE stageFlags;
         const core::smart_refctd_ptr<SamplerType>* samplers;
+
+        bool operator==(const SBinding& rhs) const
+        {
+            if (type != rhs.type)
+                return false;
+            if (count != rhs.type)
+                return false;
+            if (stageFlags != rhs.stageFlags)
+                return false;
+
+            if (!samplers && !rhs.samplers)
+                return true;
+            else if ((samplers && !rhs.samplers) || (!samplers && rhs.samplers))
+                return false;
+
+            for (uint32_t i = 0u; i < count; ++i)
+                if (samplers[i]->getParams() != rhs.samplers[i]->getParams())
```

Comment:
just compare pointers

Reply:
or does Vulkan spec say otherwise?

Reply:
vulkan just says that descriptor sets has to be "identically defined" and not sure what this means. But you might be right that immutable samplers names has to be the same

Reply:
I think comparing handles is more strict so less likely to run into problems that way.

Reply:
resolved